src/main/java/org/alliancegenome/curation_api/jobs/executors/Gff3Executor.java

```diff
@@ -103,41 +124,46 @@
+		previousIds.put("TranscriptCodingSequenceAssociation", transcriptCdsService.getIdsByDataProvider(dataProvider));
 		
 		
 		return previousIds;
 	}
 	
-	private Map<String, List<Long>> runLoad(BulkLoadFileHistory history, List<String> gffHeaderData, List<Gff3DTO> gffData,
+	private Map<String, List<Long>> runLoad(BulkLoadFileHistory history, List<String> gffHeaderData, List<ImmutablePair<Gff3DTO, Map<String,String>>> gffData,
 			Map<String, List<Long>> idsAdded, BackendBulkDataProvider dataProvider) {
 		return runLoad(history, gffHeaderData, gffData, idsAdded, dataProvider, null);
 	}
 
-	private Map<String, List<Long>> runLoad(BulkLoadFileHistory history, List<String> gffHeaderData, List<Gff3DTO> gffData,
+	private Map<String, List<Long>> runLoad(BulkLoadFileHistory history, List<String> gffHeaderData, List<ImmutablePair<Gff3DTO, Map<String,String>>> gffData,
 			Map<String, List<Long>> idsAdded, BackendBulkDataProvider dataProvider, String assemblyId) {
 		
 		ProcessDisplayHelper ph = new ProcessDisplayHelper();
 		ph.addDisplayHandler(loadProcessDisplayService);
-		ph.startProcess("GFF update for " + dataProvider.name(), (gffData.size() * 2) + 1);
+		ph.startProcess("GFF update for " + dataProvider.name(), (gffData.size() * 3) + 1);
 		
 		assemblyId = loadGenomeAssembly(assemblyId, history, gffHeaderData, dataProvider, ph);
 		idsAdded = loadEntities(history, gffData, idsAdded, dataProvider, ph);
-		idsAdded = loadAssociations(history, gffData, idsAdded, dataProvider, assemblyId, ph);
+		
+		Map<String, String> geneIdCurieMap = gff3Service.getIdCurieMap(gffData, dataProvider);
+		
+		idsAdded = loadLocationAssociations(history, gffData, idsAdded, dataProvider, assemblyId, geneIdCurieMap, ph);
+		idsAdded = loadParentChildAssociations(history, gffData, idsAdded, dataProvider, assemblyId, geneIdCurieMap, ph);
+
+		ph.finishProcess();
 		
 		return idsAdded;
 	}
 	
 	public APIResponse runLoadApi(String dataProviderName, String assemblyName, List<Gff3DTO> gffData) {
-		Map<String, List<Long>> idsAdded = new HashMap<String, List<Long>>();
-		idsAdded.put("Transcript", new ArrayList<Long>());
-		idsAdded.put("Exon", new ArrayList<Long>());
-		idsAdded.put("CodingSequence", new ArrayList<Long>());
-		idsAdded.put("TranscriptGenomicLocationAssociation", new ArrayList<Long>());
-		idsAdded.put("ExonGenomicLocationAssociation", new ArrayList<Long>());
-		idsAdded.put("CodingSequenceGenomicLocationAssociation", new ArrayList<Long>());
-		BulkLoadFileHistory history = new BulkLoadFileHistory((gffData.size() * 2) + 1);
+		Map<String, List<Long>> idsAdded = createIdsAddedMap();
 		BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(dataProviderName);
-		runLoad(history, null, gffData, idsAdded, dataProvider, assemblyName);
+		List<ImmutablePair<Gff3DTO, Map<String,String>>> preProcessedGffData = preProcessGffData(gffData, dataProvider);
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.
<!-- __reviewdog__:ChA4YTJhZmIzY2RmMzg1NWQ5EgpjaGVja3N0eWxl -->